src/components/NewGalaxyModal.js

```diff
@@ -250,3 +322,59 @@ export const NewGalaxyModal = _.flow(
     loading && spinnerOverlay
   ])
 })
+
+const MachineSelector = ({ value, onChange }) => {
+  const { cpu: currentCpu, memory: currentMemory } = findMachineType(value.machineType)
+  return h(Fragment, [
+    h(IdContainer, [
+      id => h(Fragment, [
+        label({ htmlFor: id, style: styles.label }, ['Nodes']),
+        div([
+          h(NumberInput, {
+            id,
+            min: 1,
+            max: maxNodepoolSize,
+            isClearable: false,
+            onlyInteger: true,
+            value: value.numNodes,
+            onChange: n => onChange(prevState => { return { ...prevState, numNodes: n } })
```

Comment:
Found out that updating JS objects with multiple (nested or not) fields works differently for function components than class components. For the latter, if only one field of such a JS object is updated, it gets merged as desired. For the former, one has to update the entire object, which is why I used the spread operator (`…`) as suggested by [this blog](https://blog.logrocket.com/a-guide-to-usestate-in-react-ecb9952e406c).